File: hao-yang-finance-app/src/features/Settings/types/driver.ts
export interface Driver {
	id: string;
	name: string;
	phone?: string;
	profitShareRatio?: number;
	truckTonnage?: number;
	isActive: boolean;
	createdAt: string;
	updatedAt: string;
}

export interface CreateDriverDto {
	name: string;
	phone?: string;
	profitShareRatio?: number;
	truckTonnage?: number;
}

export interface UpdateDriverDto {
	name: string;
	phone?: string;
	profitShareRatio?: number;
	truckTonnage?: number;
	isActive: boolean;
}

File: hao-yang-finance-api/Models/Driver.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hao_yang_finance_api.Models
{
    [Table("driver")]
    public class Driver
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Column("name")]
        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        [Column("phone")]
        [MaxLength(20)]
        public string? Phone { get; set; }

        [Column("profit_share_ratio")]
        public decimal ProfitShareRatio { get; set; } = 50;

        [Column("truck_tonnage")]
        public decimal? TruckTonnage { get; set; }

        [Column("is_active")]
        public bool IsActive { get; set; } = true;

        [Column("created_at")]
        public string CreatedAt { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ");

        [Column("updated_at")]
        public string UpdatedAt { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ");

        // Navigation properties
        public virtual ICollection<Waybill> Waybills { get; set; } = new List<Waybill>();
        public virtual ICollection<DriverSettlement> Settlements { get; set; } = new List<DriverSettlement>();
    }
}